src/engine/strat_engine/blockdev.rs

```diff
@@ -76,30 +74,29 @@ pub fn find_all() -> EngineResult<BTreeMap<PoolUuid, BTreeMap<DevUuid, BlockDev>
                                format!("Could not open {}", devnode.display()))
             }));
 
-        let mut buf = [0u8; 4096];
+        let mut buf = [0u8; BDA_STATIC_HDR_SIZE as usize];
```

Comment:
These casts are potential trouble wherever they happen. What it be any help to make these two constants usize?